packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -36,14 +36,21 @@ const DropdownItem = ({ className, children, isHovered, component: Component, is
   } else if (Component === 'button') {
     additionalProps.disabled = isDisabled;
   }
+
   return (
     <li>
-      <Component
-        {...additionalProps}
-        className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
-      >
-        {children}
-      </Component>
+      {React.isValidElement(children) ? (
+        React.Children.map(children, child =>
+          React.cloneElement(child, { className: `${className} ${child.props.className}` })
```

Comment:
So that was my initial cause for saying that if just className is copied over then we might as well skip the cloning. I guess for this to provide our PF styling here, I would expect to see something like this here:
```
React.cloneElement(child, { 
          className: css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className) 
        }
```

Reply:
Ah I forgot about the disabled and hovered styles, I can add those. It's a bit confusing, but className is actually the PF class being applied, child.props.className is whatever was manually declared as a className in the Link itself. Or should I not keep the local class?

Reply:
Yes that sounds good to add the local class as well!